src/hoc/withTheme.js

```diff
@@ -32,17 +33,19 @@ const wrapWithTheme = (Component: ReactClass<any>) => {
 
     componentWillMount() {
       const styledContext = this.context[CHANNEL_NEXT]
-      if (styledContext === undefined) {
+      if (!styledContext && !this.constructor.defaultProps) {
         // eslint-disable-next-line no-console
-        console.error('[withTheme] Please use ThemeProvider to be able to use withTheme')
-        return
-      }
-
-
-      const { subscribe } = styledContext
-      this.unsubscribeId = subscribe(theme => {
+        console.warn('[withTheme] You are not using a ThemeProvider nor passing a theme in defaultProps')
```

Comment:
This warning has incorrect conditions (defaultProps is set) and an inaccurate message now.

Can you change it to show up when the styles context is not present and the result of the util is undefined?

The message will also have to change to something that indicates that no themeprovider was used, no theme prop was passed, and no default theme prop is present

This will also only be necessary in development mode as a warning now since it’s pretty predictable and deterministic behaviour